feat(book): add validation and global exception handling

vDmytriv01 opened the pull request:
- Added global exception handler (CustomGlobalExceptionHandler)
- Added PriceRangeValidation and PriceRangeValidator for minPrice/maxPrice checks
- Updated CreateBookRequestDto with NotBlank and Min constraints
- Updated BookSearchParametersDto with custom price range validation
- Updated BookController to use Valid for request DTOs
- Added validation dependency to pom.xml

vDmytriv01 committed:
feat(book): add validation and global exception handling

- Added global exception handler (CustomGlobalExceptionHandler)
- Added PriceRangeValidation and PriceRangeValidator for minPrice/maxPrice checks
- Updated CreateBookRequestDto with @NotBlank and @Min constraints
- Updated BookSearchParametersDto with custom price range validation
- Updated BookController to use @Valid for request DTOs
- Added validation dependency to pom.xml

liliia-ponomarenko commented on src/main/java/com/vdmytriv/bookstoreapp/dto/UpdateBookRequestDto.java:
```diff
@@ -1,12 +1,15 @@
+        @Min(0)
+        BigDecimal price,
+        String description,
+        String coverImage
```
Should you validate other fields? Do you use the full update or only the provided fields?

liliia-ponomarenko commented on src/main/java/com/vdmytriv/bookstoreapp/exception/CustomGlobalExceptionHandler.java:
```diff
@@ -0,0 +1,63 @@
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.INTERNAL_SERVER_ERROR.value());
+        body.put("error", ex.getMessage());
```
Can you move the common logic to the private method?

liliia-ponomarenko reviewed (approved):
Good job! A few recommendations ;)

vDmytriv01 committed:
refactor: applied the advice of the mentor

- Added PATCH /books/{id} endpoint for partial book updates
- Created PartialUpdateBookRequestDto with Optional fields
- Added @NotBlank and @Pattern annotations to UpdateBookRequestDto
- Implemented BookService.patch() and partialUpdateModel() in BookMapper
- Refactored exception handlers to use buildErrorResponse() for consistent error formatting
- Added handler for DataIntegrityViolationException returning BAD_REQUEST with root cause message
